### ISSUE TITLE
Revoke standard delegate from pNFT

### ISSUE THREAD

**Issue by tsf-sol** [metaplex-foundation/metaplex-program-library#1105]
Hello,

In reference to this line of code: https://github.com/metaplex-foundation/metaplex-program-library/blob/ae436e9734977773654fb8ea0f72e3ac559253b8/token-metadata/program/src/processor/delegate/revoke.rs#L217

it makes no sense to raise an error here, as we have NFTs that were locked with StandardV1 and got upgraded to pNFT and are currently locked forever due to this line.
You either need to raise the error on migration, or on adding the delegate, but not when removing it.

Can you please remove this line since it doesn't affect anything in your delegation role ?

**Comment by samuelvanderwaal** [comment]
> Hello,
> 
> In reference to this line of code:
> 
> https://github.com/metaplex-foundation/metaplex-program-library/blob/ae436e9734977773654fb8ea0f72e3ac559253b8/token-metadata/program/src/processor/delegate/revoke.rs#L217
> 
> it makes no sense to raise an error here, as we have NFTs that were locked with StandardV1 and got upgraded to pNFT and are currently locked forever due to this line. You either need to raise the error on migration, or on adding the delegate, but not when removing it.
> 
> Can you please remove this line since it doesn't affect anything in your delegation role ?

Can you give me the mint address of one of these NFTs? All migrated NFTs should be upgraded with a `MigrationDelegate`.

**Comment by tsf-sol** [comment]
> 

- AdtZyBhtBCFRbXGjqtnrchgxBtXNW7Kzqomk67oD7rn9
- 5QfQRmuveCPq7Kp93PtZDHjwvwoTKckuCpAYPFLQ8G1i
- BZJEJngX89bxrHjBeLzmW7VcFNPPwDMPzCqVdknwcWoT

These are 3 examples, however, we fixed it for people with the following way:
1- unlock nft with `createUnlockInstruction`
2- transfer to TSF wallet with `createTransferInstruction`
3- close user nft account with `createCloseAccountInstruction`
4- transfer to user wallet with `createTransferInstruction`
5- close TSF nft account with `createCloseAccountInstruction`

this way, a new Token Account will be initialized on the user's end and it won't have the delegate in it.

**Comment by samuelvanderwaal** [comment]
Gotcha. This is a weird edge-case bug which we recently fixed and merged into master but haven't deployed the fix yet. It's supposed to go out today.

**Comment by samuelvanderwaal** [comment]
This fix is on mainnet now, let me know if you encounter this issue in the future.

**Comment by tsf-sol** [comment]
> Gotcha. This is a weird edge-case bug which we recently fixed and merged into master but haven't deployed the fix yet. It's supposed to go out today.

Hey Sam, yes, we're in talk with Verata too and he told us about the fix. I can test out the previous version of the code if it's still working or no, but will need to find someone who's having this issue which will be hard to do with the big community that we have :')

**Comment by samuelvanderwaal** [comment]
> > Gotcha. This is a weird edge-case bug which we recently fixed and merged into master but haven't deployed the fix yet. It's supposed to go out today.
> 
> Hey Sam, yes, we're in talk with Verata too and he told us about the fix. I can test out the previous version of the code if it's still working or no, but will need to find someone who's having this issue which will be hard to do with the big community that we have :')

No worries. If you're unblocked I'm happy, you don't need to try to find an example to test the fix. Just let me know if you run into the issue again (you shouldn't, Verata said the fix is working).
